src/schema/posts.ts

```diff
@@ -1582,6 +1635,91 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
     },
   },
   Mutation: {
+    createSourcePostModeration: async (
+      _,
+      {
+        sourceId,
+        image,
+        title,
+        content,
+        commentary,
+        type,
+        sharedPostId,
+        imageUrl,
+        externalLink,
+      }: CreateSourcePostModerationArgs,
+      ctx: AuthContext,
+      info,
+    ): Promise<GQLSourcePostModeration> => {
+      const { con, userId } = ctx;
+
+      const sourceMember = await con.getRepository(SourceMember).findOne({
+        where: { sourceId: sourceId, userId: userId },
+        relations: ['source'],
+      });
+
+      const source = await sourceMember?.source;
+
+      if (!source) {
+        throw new ForbiddenError('Access denied!');
+      }
+
+      await canAccessSource(
+        ctx,
+        source,
+        sourceMember,
+        SourcePermissions.Post,
+        sourceId,
+      );
+      const id = generateUUID();
+      const mentions = await getMentions(con, content, userId, sourceId);
+
+      const pendingPost: CreateSourcePostModeration = {
+        id,
+        title,
+        content,
+        sourceId,
+        type,
+        sharedPostId,
+        externalLink,
+        createdById: userId,
+      };
+
+      if (commentary) {
+        await validateCommentary(commentary);
+        pendingPost.titleHtml = generateTitleHtml(commentary, mentions);
+      }
+      if (content) {
+        pendingPost.contentHtml = markdown.render(content, { mentions });
+      }
+
+      await con.transaction(async (manager) => {
+        if (imageUrl) {
+          pendingPost.image = imageUrl;
+        } else if (image && process.env.CLOUDINARY_URL) {
+          const upload = await image;
+          const { url: coverImageUrl } = await uploadPostFile(
+            id,
+            upload.createReadStream(),
+            UploadPreset.PostBannerImage,
+          );
+          pendingPost.image = coverImageUrl;
+        }
+        await createSourcePostModeration(manager, pendingPost);
+      });
+
+      return graphorm.queryOneOrFail<GQLSourcePostModeration>(
```

Comment:
If we going to query anyway, you might as well just get the return value from the insert?

Reply:
We do this on other post creation as well. Asked @capJavert about it and the reason we do it this way is to retain the GraphQL schema

Reply:
Yeah aware of that, meant more you could maybe return the inserted ID, then return the graphORM layer?